index.js

```diff
@@ -26,8 +26,24 @@ maths.import({
 
 function transformResolve(value) {
     return reduceFunctionCall(value, 'resolve', function(argString) {
+
+        var unit = '';
+        if (argString.indexOf('floor(') >= 0
```

Comment:
It will be better to support multiple floor/ceiling but I am not sure to
understand how to update the Unit prototype.
It should be in mathjs to support floor and ceil with Unit and not only
number? I will be great but I don't known if they do not wish support it
(exclicit test 'should throw an error with a unit' checking the no support:
https://github.com/josdejong/mathjs/blob/d065b04033ef9fa1dd295d3f76bd06b20f752c89/test/function/arithmetic/floor.test.js
)

On Wed, Jul 29, 2015 at 3:45 PM, Shaun Stanworth <notifications@github.com>
wrote:

> In index.js
> <https://github.com/shauns/postcss-math/pull/2#discussion_r35759730>:
>
> > @@ -26,8 +26,24 @@ maths.import({
> >
> >  function transformResolve(value) {
> >      return reduceFunctionCall(value, 'resolve', function(argString) {
> > +
> > +        var unit = '';
> > +        if (argString.indexOf('floor(') >= 0
>
> Looks good, but I wonder if we could move this logic and adapt the Unit
> prototype instead maybe? This'll mean we can support multiple floor/ceiling
> calls in one resolve string. Open to your preference [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/shauns/postcss-math/pull/2/files#r35759730>.
>